chromium_src/third_party/blink/renderer/core/frame/location.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/frame/location.h"
+
+#include "base/ranges/algorithm.h"
+
+#define ancestorOrigins ancestorOrigins_ChromiumImpl
+
+#include "src/third_party/blink/renderer/core/frame/location.cc"
+
+#undef ancestorOrigins
+
+namespace blink {
+
+DOMStringList* Location::ancestorOrigins() const {
+  DOMStringList* origins = ancestorOrigins_ChromiumImpl();
+
+  auto is_chrome_untrusted = [](const String& origin) {
+    return origin.StartsWith("chrome-untrusted://");
```

Comment:
I wonder if that looks fine or it's better to be explicit here and do the filtering only for `chrome-untrusted://player-embed`.

Reply:
We may as well remove all `chrome-untrusted://` URLs from here. That special scheme is not really meant to be exposed to websites.

Reply:
sounds good